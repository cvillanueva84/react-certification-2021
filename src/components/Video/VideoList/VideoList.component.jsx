import React from 'react';
import { videoData } from '../../../youtube-videos-mock';
import { Container } from './VideoList.styled';
import VideoItem from '../VideoItem/VideoItem.component';

export default function VideoList() {
  const { items } = videoData;

  const avatar = items[0].snippet.thumbnails.default.url;

  return (
    <Container>
      {items.slice(1).map((item, id) => {
        const {
          snippet: {
            thumbnails: {
              medium: { url },
            },
          },
        } = item;

        const {
          snippet: { title },
        } = item;

        const {
          snippet: { description },
        } = item;

        return (
          <VideoItem
            key={id}
            url={url}
            title={title}
            description={description}
            avatar={avatar}
          />
        );
      })}
    </Container>
  );
}
