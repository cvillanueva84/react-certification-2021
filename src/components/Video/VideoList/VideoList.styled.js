import styled from 'styled-components';

export const Container = styled.section`
  margin-top: 100px;
  position: relative;
  z-index: 0;
  width: 800px;

  @media screen and (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 1.5rem;
  }

  @media screen and (min-width: 992px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;
