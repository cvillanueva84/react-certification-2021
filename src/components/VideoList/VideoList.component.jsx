import React from 'react';
import styled from 'styled-components';

const VideoCardDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 30rem;
  height: 35rem;
  border-radius: 0.4rem;
  margin: 2rem;
  border: 1px #dad4d4 solid;
  transition: all 0.3s ease;
  &:hover {
    box-shadow: 10px 10px 24px 0px rgba(0, 0, 0, 0.75);
  }

  @media only screen and (max-width: 670px) {
    width: 80%;
  }
`;
VideoCardDiv.displayName = 'VideoCardDiv';
const VideoImageDiv = styled.div`
  height: 140px;
`;
VideoImageDiv.displayName = 'VideoImageDiv';
const TitleSpan = styled.span`
  font-size: 1.6rem;
  font-weight: bold;
  margin: 2rem;
`;
TitleSpan.displayName = 'TitleSpan';

const DescriptionSpan = styled.span`
  font-size: 1.4rem;
  margin-bottom: 2rem;
  text-align: left;
  margin: 2rem;
`;
DescriptionSpan.displayName = 'DescriptionSpan';

const VideoList = ({ video }) => {
  const { title, description, thumbnails } = video;

  return (
    <VideoCardDiv>
      <VideoImageDiv
        style={{
          backgroundImage: `url(${thumbnails.medium.url})`,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
          backgroundPosition: 'center',
          borderTopRightRadius: '0.4rem',
          borderTopLeftRadius: '0.4rem',
        }}
      ></VideoImageDiv>
      <TitleSpan>{title}</TitleSpan>
      <DescriptionSpan>{description ? description : 'watch video...'}</DescriptionSpan>
    </VideoCardDiv>
  );
};

export default VideoList;
