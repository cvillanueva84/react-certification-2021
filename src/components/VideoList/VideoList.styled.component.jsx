/* eslint-disable import/no-extraneous-dependencies */
import styled from 'styled-components';

export const VideoListContainer = styled.section`
  /* background-color: #eb71b8; */
  max-width: 100%;
  max-height: 100%;
  display: flex;
  align-self: flex-evenly;
  flex-wrap: wrap;
  padding: 0 24px;
  /* box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%), */
  /*   0px 1px 10px 0px rgb(0 0 0 / 12%); */
`;

export const VideoListTitle = styled.h2`
  font-size: 2rem;
  letter-spacing: -2px;
  text-align: center;
  width: 100%;
`;

export const Videos = styled.div`
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
`;
