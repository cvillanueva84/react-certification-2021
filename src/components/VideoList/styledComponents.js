import styled from 'styled-components';

export const VideoCardDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 30rem;
  height: 35rem;
  border-radius: 0.4rem;
  margin: 2rem;
  border: 1px #dad4d4 solid;
  transition: all 0.3s ease;
  &:hover {
      box-shadow: 10px 10px 24px 0px rgba(0, 0, 0, 0.75);
    }
    
    @media only screen and (max-width: 670px) {
        width: 80%;
    }
`;
VideoCardDiv.displayName = 'VideoCardDiv';

export const VideoImageDiv = styled.div`
  height: 140px;
`;
VideoImageDiv.displayName = 'VideoImageDiv';

export const TitleSpan = styled.span`
  font-size: 1.6rem;
  font-weight: bold;
  margin: 2rem;
`;
TitleSpan.displayName = 'TitleSpan';

export const DescriptionSpan = styled.span`
  font-size: 1.4rem;
  margin-bottom: 2rem;
  text-align: left;
  margin: 2rem;
  `;
DescriptionSpan.displayName = 'DescriptionSpan';

