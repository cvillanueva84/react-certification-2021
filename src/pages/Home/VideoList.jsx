import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 