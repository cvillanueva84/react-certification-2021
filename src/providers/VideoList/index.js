export {  VideoListProvider } from './VideoList.provider';
