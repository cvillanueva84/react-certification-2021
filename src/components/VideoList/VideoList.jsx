import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
