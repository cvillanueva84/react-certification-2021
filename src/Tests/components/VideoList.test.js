import React from 'react';
import { shallow } from 'enzyme';
import VideoList from '../../components/VideoList/VideoList.component';

describe('Tests of the VideoList component', () => {

    const data =  {
        "kind": "youtube#searchResult",
        "etag": "_PVKwNJf_qw9nukFeRFOtQ837o0",
        "id": {
          "kind": "youtube#channel",
          "videoId": "UCPGzT4wecuWM0BH9mPiulXg"
        },
        "snippet": {
          "publishedAt": "2014-09-27T01:39:18Z",
          "channelId": "UCPGzT4wecuWM0BH9mPiulXg",
          "title": "Wizeline",
          "description": "Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...",
          "thumbnails": {
            "default": {
              "url": "https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0xffffffff-no-rj-mo"
            },
            "medium": {
              "url": "https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
            },
            "high": {
              "url": "https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
            }
          },
          "channelTitle": "Wizeline",
          "liveBroadcastContent": "upcoming",
          "publishTime": "2014-09-27T01:39:18Z"
        }
      }
    
    const wrapper = shallow(< VideoList video={data}/>)
    test('should render the VideoList component', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    
    test('should contain the VideoCardDiv styled component', () => {
        expect(wrapper.find('VideoCardDiv').length).toEqual(1);
    })

    test('should contain the VideoImageDiv styled component', () => {
        expect(wrapper.find('VideoImageDiv').length).toEqual(1);
    })

    test('should contain the TitleSpan styled component', () => {
        expect(wrapper.find('TitleSpan').length).toEqual(1);
    })

    test('should contain the DescriptionSpan styled component', () => {
        expect(wrapper.find('DescriptionSpan').length).toEqual(1);
    })

    test('VideoImageDiv background Image should be the thumbnails.medium.url prop', () => {
        const imgDiv = wrapper.find('VideoImageDiv')
        expect(imgDiv.prop('style').backgroundImage).toBe(`url(${data.snippet.thumbnails.medium.url})`)
    })
    

    test('Title Span styled component should display the title prop', () => {
        const titleSpan = wrapper.find('TitleSpan')
        expect(titleSpan.text()).toBe(data.snippet.title)
    })

    test('Title Span styled component should display the title prop', () => {
        const titleSpan = wrapper.find('TitleSpan')
        expect(titleSpan.text()).toBe(data.snippet.title)
    })
    
    test('Description Span styled component should display the description prop', () => {
        const descriptionSpan = wrapper.find('DescriptionSpan')
        expect(descriptionSpan.text()).toBe(data.snippet.description)
    })
})

