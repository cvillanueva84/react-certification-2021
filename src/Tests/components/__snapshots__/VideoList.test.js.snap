// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the VideoList component should render the VideoList component 1`] = `
<VideoCardDiv>
  <VideoImageDiv
    style={
      Object {
        "backgroundImage": "url(https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
      }
    }
  />
  <TitleSpan>
    Wizeline
  </TitleSpan>
  <DescriptionSpan>
    Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
  </DescriptionSpan>
</VideoCardDiv>
`;
